alphadia/search_step.py

```diff
@@ -271,7 +271,7 @@ def get_run_data(self) -> Generator[tuple[str, str, SpecLibFlat]]:
         """Generator for raw data and spectral library."""
 
         if self.spectral_library is None:
-            raise ValueError("no spectral library loaded")
+            raise NoLibraryAvailableError()
```

Comment:
If we raise here, the error message should say that there needs to be a library or library prediction as mentioned. 

Otherwise, we could also raise on the level of the search plan with individual errors.